### PR TITLE
Add a tree revision lookup API to NodeReader

### PR DESCRIPTION
**Pull request by Martin2112**
Add a tree revision lookup API to NodeReader + tests

### CODE REVIEW

File: storage/mysql/tree_storage.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ const selectLatestSignedRootSql string = `SELECT TreeHeadTimestamp,TreeSize,Root
 		 ORDER BY TreeHeadTimestamp DESC LIMIT 1`
 const insertTreeHeadSql string = `INSERT INTO TreeHead(TreeId,TreeHeadTimestamp,TreeSize,RootHash,TreeRevision,RootSignature)
 		 VALUES(?,?,?,?,?,?)`
+const selectTreeRevisionAtSizeSql string = "SELECT TreeRevision FROM TreeHead WHERE TreeId=? AND TreeSize=?"
```
Comment (AlCutter) [inline storage/mysql/tree_storage.go]: There could be multiple result rows returned by this if nothing is added to the tree between multiple signings, I'd expect they should all have the same `TreeRevision` though.

File: storage/mysql/tree_storage.go [hunk 2]
```diff
@@ -711,6 +712,20 @@ func (t *tx) removeSequencedLeaves(leaves []trillian.LogLeaf) error {
 	return nil
 }
 
+// TODO: This only works for sizes where there is a stored tree head. This is deliberate atm
+// as serving proofs at intermediate tree sizes is complicated and will be implemented later.
+func (t *tx) GetTreeRevisionAtSize(treeSize int64) (int64, error) {
```
Comment (AlCutter) [inline storage/mysql/tree_storage.go]: API godoc missing on the new funcs in here

File: storage/mysql/tree_storage.go (hunk 1 above)
Comment (Martin2112) [inline storage/mysql/tree_storage.go]: Changed it to take the highest one, though they should be the same. Added a test for this.

File: storage/mysql/tree_storage.go (hunk 2 above)
Comment (Martin2112) [inline storage/mysql/tree_storage.go]: Cut and pasted, looks like an opportunity for it to get out of date though. Inheriting docs was probably another NIH thing.

**Pull request merged**